"export" goal of querydsl-maven-plugin can't create class implements java.io.Serializable

refs #558 .

my pom.xml

```
        <plugin>
            <groupId>com.mysema.querydsl</groupId>
            <artifactId>querydsl-maven-plugin</artifactId>
            <version>3.3.0</version>
            <executions>
                <execution>
                    <goals>
                        <goal>export</goal>
                    </goals>
                </execution>
            </executions>
            <configuration>
                <jdbcDriver>org.postgresql.Driver</jdbcDriver>
                <!-- snip -->
                <beanInterfaces>java.io.Serializable</beanInterfaces>
                <exportBeans>true</exportBeans>
                <targetFolder>${project.basedir}/target/generated-sources/java</targetFolder>
            </configuration>
```

expected result: get the table bean source code like below:

  pubilc class Foo implements java.io.Serializable { ...

actual result:

  [INFO] ------------------------------------------------------------------------
  [ERROR] BUILD ERROR
  [INFO] ------------------------------------------------------------------------
  [INFO] Failed to configure plugin parameters for: com.mysema.querydsl:querydsl-maven-plugin:3.3.0

```
(found static expression: 'java.io.Serializable' which may act as a default value).
```

  Cause: Cannot assign configuration entry 'beanInterfaces' to 'class [Ljava.lang.String;' from 'java.io.Serializable', which is of type class java.lang.String

--- a/querydsl-codegen/src/main/java/com/mysema/query/codegen/BeanSerializer.java
+++ b/querydsl-codegen/src/main/java/com/mysema/query/codegen/BeanSerializer.java
@@ -13,25 +13,15 @@
  */
 package com.mysema.query.codegen;
 
+import javax.annotation.Generated;
 import java.io.IOException;
 import java.lang.annotation.Annotation;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-import javax.annotation.Generated;
+import java.util.*;
 
 import com.google.common.base.Function;
 import com.google.common.collect.Lists;
 import com.mysema.codegen.CodeWriter;
-import com.mysema.codegen.model.ClassType;
-import com.mysema.codegen.model.Parameter;
-import com.mysema.codegen.model.Type;
-import com.mysema.codegen.model.TypeCategory;
-import com.mysema.codegen.model.Types;
+import com.mysema.codegen.model.*;
 import com.mysema.util.BeanUtils;
 
 /**
@@ -52,7 +42,7 @@ public class BeanSerializer implements Serializer{
 
     private final boolean propertyAnnotations;
 
-    private final List<Class<?>> interfaces = Lists.newArrayList();
+    private final List<Type> interfaces = Lists.newArrayList();
 
     private final String javadocSuffix;
 
@@ -108,8 +98,8 @@ public class BeanSerializer implements Serializer{
 
         // imports
         Set<String> importedClasses = getAnnotationTypes(model);
-        for (Class<?> iface : interfaces) {
-            importedClasses.add(iface.getName());
+        for (Type iface : interfaces) {
+            importedClasses.add(iface.getFullName());
         }
         importedClasses.add(Generated.class.getName());
         if (model.hasLists()) {
@@ -144,11 +134,8 @@ public class BeanSerializer implements Serializer{
             if (printSupertype && model.getSuperType() != null) {
                 superType = model.getSuperType().getType();
             }
-            Type[] interfaceTypes = new Type[interfaces.size()];
-            for (int i = 0; i < interfaceTypes.length; i++) {
-                interfaceTypes[i] = new ClassType(interfaces.get(i));
-            }
-            writer.beginClass(model, superType, interfaceTypes);
+            Type[] ifaces = interfaces.toArray(new Type[interfaces.size()]);
+            writer.beginClass(model, superType, ifaces);
         } else if (printSupertype && model.getSuperType() != null) {
             writer.beginClass(model, model.getSuperType().getType());
         } else {
@@ -253,7 +240,11 @@ public class BeanSerializer implements Serializer{
     }
 
     public void addInterface(Class<?> iface) {
-        interfaces.add(iface);
+        interfaces.add(new ClassType(iface));
+    }
+
+    public void addInterface(Type type) {
+        interfaces.add(type);
     }
 
     public void setAddToString(boolean addToString) {

--- a/querydsl-codegen/src/test/java/com/mysema/query/codegen/BeanSerializerTest.java
+++ b/querydsl-codegen/src/test/java/com/mysema/query/codegen/BeanSerializerTest.java
@@ -13,9 +13,6 @@
  */
 package com.mysema.query.codegen;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-
 import java.io.IOException;
 import java.io.Serializable;
 import java.io.StringWriter;
@@ -23,16 +20,13 @@ import java.io.Writer;
 import java.util.Arrays;
 import java.util.Date;
 
-import org.junit.Before;
-import org.junit.Test;
-
 import com.mysema.codegen.JavaWriter;
 import com.mysema.codegen.StringUtils;
-import com.mysema.codegen.model.ClassType;
-import com.mysema.codegen.model.SimpleType;
-import com.mysema.codegen.model.Type;
-import com.mysema.codegen.model.TypeCategory;
-import com.mysema.codegen.model.Types;
+import com.mysema.codegen.model.*;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
 
 public class BeanSerializerTest {
 
@@ -100,6 +94,14 @@ public class BeanSerializerTest {
 
     @Test
     public void Interfaces() throws IOException {
+        BeanSerializer serializer = new BeanSerializer();
+        serializer.addInterface(new ClassType(Serializable.class));
+        serializer.serialize(type, SimpleSerializerConfig.DEFAULT, new JavaWriter(writer));
+        assertTrue(writer.toString().contains("public class DomainClass implements Serializable {"));
+    }
+
+    @Test
+    public void Interfaces2() throws IOException {
         BeanSerializer serializer = new BeanSerializer();
         serializer.addInterface(Serializable.class);
         serializer.serialize(type, SimpleSerializerConfig.DEFAULT, new JavaWriter(writer));

--- a/querydsl-maven-plugin/src/main/java/com/mysema/query/maven/AbstractMetaDataExportMojo.java
+++ b/querydsl-maven-plugin/src/main/java/com/mysema/query/maven/AbstractMetaDataExportMojo.java
@@ -19,11 +19,7 @@ import java.sql.DriverManager;
 import java.sql.SQLException;
 import java.util.Comparator;
 
-import org.apache.maven.plugin.AbstractMojo;
-import org.apache.maven.plugin.MojoExecutionException;
-import org.apache.maven.plugin.MojoFailureException;
-import org.apache.maven.project.MavenProject;
-
+import com.mysema.codegen.model.SimpleType;
 import com.mysema.query.codegen.BeanSerializer;
 import com.mysema.query.sql.Configuration;
 import com.mysema.query.sql.SQLTemplates;
@@ -31,6 +27,10 @@ import com.mysema.query.sql.codegen.DefaultNamingStrategy;
 import com.mysema.query.sql.codegen.MetaDataExporter;
 import com.mysema.query.sql.codegen.NamingStrategy;
 import com.mysema.query.sql.types.Type;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.project.MavenProject;
 
 /**
  * MetaDataExportMojo is a goal for MetaDataExporter usage
@@ -343,7 +343,14 @@ public class AbstractMetaDataExportMojo extends AbstractMojo{
                     BeanSerializer serializer = new BeanSerializer();
                     if (beanInterfaces != null) {
                         for (String iface : beanInterfaces) {
-                            serializer.addInterface(Class.forName(iface));
+                            int sepIndex = iface.lastIndexOf('.');
+                            if (sepIndex < 0) {
+                                serializer.addInterface(new SimpleType(iface));
+                            } else {
+                                String packageName = iface.substring(0, sepIndex);
+                                String simpleName = iface.substring(sepIndex + 1);
+                                serializer.addInterface(new SimpleType(iface, packageName, simpleName));
+                            }
                         }
                     }
                     serializer.setAddFullConstructor(beanAddFullConstructor);

--- a/querydsl-sql-codegen/src/main/java/com/mysema/query/sql/ant/AntMetaDataExporter.java
+++ b/querydsl-sql-codegen/src/main/java/com/mysema/query/sql/ant/AntMetaDataExporter.java
@@ -19,13 +19,13 @@ import java.sql.DriverManager;
 import java.sql.SQLException;
 import java.util.Comparator;
 
-import org.apache.tools.ant.BuildException;
-import org.apache.tools.ant.Task;
-
+import com.mysema.codegen.model.SimpleType;
 import com.mysema.query.codegen.BeanSerializer;
 import com.mysema.query.sql.codegen.DefaultNamingStrategy;
 import com.mysema.query.sql.codegen.MetaDataExporter;
 import com.mysema.query.sql.codegen.NamingStrategy;
+import org.apache.tools.ant.BuildException;
+import org.apache.tools.ant.Task;
 
 /**
  * AntMetaDataExporter exports JDBC metadata to Querydsl query types
@@ -248,10 +248,13 @@ public class AntMetaDataExporter extends Task {
                 BeanSerializer serializer = new BeanSerializer();
                 if (beanInterfaces != null) {
                     for (String iface : beanInterfaces) {
-                        try {
-                            serializer.addInterface(Class.forName(iface));
-                        } catch (ClassNotFoundException e) {
-                            throw new BuildException(e.getMessage(), e);
+                        int sepIndex = iface.lastIndexOf('.');
+                        if (sepIndex < 0) {
+                            serializer.addInterface(new SimpleType(iface));
+                        } else {
+                            String packageName = iface.substring(0, sepIndex);
+                            String simpleName = iface.substring(sepIndex + 1);
+                            serializer.addInterface(new SimpleType(iface, packageName, simpleName));
                         }
                     }
                 }

